react-accordion-component/src/accordion.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+
+function Elements(props) {
+  const id = props.id;
+
+  if (props.element) {
+    return (
+    <div className='topic-container' onClick={e => props.handleTrue(id)}>
+      <div className="topic-header">
+        <li>{props.name}</li>
+      </div>
+      <div className="topic-info hidden">
+        {props.info}
+      </div>
+    </div>
+    );
+  } else {
+    return (
+    <div className='topic-container' onClick={e => props.handleFalse(id)}>
+      <div className="topic-header">
+        <li>{props.name}</li>
+      </div>
+      <div className="topic-info">
+        {props.info}
+      </div>
+    </div>
+    );
+  }
+}
+
+class Accordion extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      1: true,
+      2: true,
+      3: true
+    };
```

Comment:
The issue with this approach is that it requires the accordion to know how many topics it's handling in advance, so it's less flexible. Instead, just track _one_ value, something like `openTopic` or `currentIndex`.

If you set this one value to `null`, you know everything is closed. When the user clicks a topic header, update the `currentIndex` and during your re-render, you can now decide whether to show or hide the details for that topic. 

If the user clicks the same one again, set it back to `null`.